module.nix

```diff
@@ -0,0 +1,348 @@
+      };
+      database = mkOption {
+        type = types.attrsOf types.anything;
+        description = ''
+          The database the application should connect to.
+          See https://docs.djangoproject.com/en/3.2/ref/settings/#databases for details.
+          Environment variables will be substituted at runtime.
+        '';
+        default = {
+          ENGINE = "django.db.backends.sqlite3";
+          NAME = "$STATE_DIRECTORY/db.sqlite3";
+        };
+        example = {
+          ENGINE = "django.db.backends.mysql";
+          NAME = "course_management";
+          USER = "coursemanagement";
+          PASSWORD = "$MYSQL_PASSWORD";
+          HOST = "127.0.0.1";
+          OPTIONS = {
+            charset = "utf8mb4";
+            user_unicode = true;
+          };
+        };
+      };
+      admins = mkOption {
+        type = types.listOf (types.submodule {
+          options = {
+            name = mkOption {
+              type = types.str;
+            };
+            email = mkOption {
+              type = types.str;
+            };
+          };
+        });
+        description = ''
+          A list of all the people who get code error notifications.
+        '';
+        default = [ ];
+        example = [{
+          name = "Admin";
+          email = "root@example.com";
+        }];
+      };
+      allowedHosts = mkOption {
+        type = types.listOf types.str;
+        description = "A list of hostnames that may be served.";
+        default = [ (mkIf (cfg.hostName != null) cfg.hostName) ];
+      };
+      email = mkOption {
+        type = settingsEmailType;
+        description = "Configuration for sending email.";
+        default = {
+          host = "localhost";
+          port = 25;
+          user = "";
+          passwordFile = "/dev/null";
+          fromEmail = "webmaster@localhost";
+          serverEmail = "root@localhost";
+        };
+      };
+      extraConfig = mkOption {
+        default = "";
+        type = types.lines;
+        description = ''
+          Extra configuration options that will be added verbatim at
+          the end of `settings.py`.
+        '';
+      };
+    };
+  };
+
+  settingsEmailType = with lib; types.submodule {
+    options = {
+      host = mkOption {
+        type = types.str;
+      };
+      port = mkOption {
+        type = types.int;
+      };
+      user = mkOption {
+        type = types.str;
+      };
+      passwordFile = mkOption {
+        type = types.str;
+      };
+      fromEmail = mkOption {
+        type = types.str;
+      };
+      serverEmail = mkOption {
+        type = types.str;
+        description = "Email used for error notifications.";
+      };
+    };
+  };
+in
+{
+  options.services.course-management = with lib; {
+    enable = mkEnableOption "iFSR course management";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.course-management;
+      description = "The package to use.";
+    };
```

Comment:
Maybe it's worthwile introducing a `user` option in here that we can use for sops.